src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -8008,47 +7981,42 @@ export class StudyViewPageStore
     readonly molecularProfileSampleCounts = remoteData<
         MultiSelectionTableRow[]
     >({
-        await: () => [this.molecularProfiles, this.selectedSamples],
+        await: () => [this.molecularProfiles],
         invoke: async () => {
-            // return empty if there are no filtered samples
-            if (!this.hasFilteredSamples) {
-                return [];
-            }
-            const counts = await internalClient.fetchMolecularProfileSampleCountsUsingPOST(
-                {
+            const [counts, selectedSamples] = await Promise.all([
+                internalClient.fetchMolecularProfileSampleCountsUsingPOST({
                     studyViewFilter: this.filters,
-                }
-            );
+                }),
+                toPromise(this.selectedSamples),
+            ]);
 
             return counts.map(caseListOption => {
                 return {
                     uniqueKey: caseListOption.value,
                     label: caseListOption.label,
                     numberOfAlteredCases: caseListOption.count,
-                    numberOfProfiledCases: this.selectedSamples.result.length,
+                    numberOfProfiledCases: selectedSamples.length,
```

Comment:
is it odd that we assume all seelctedSamples are profiled here?

Reply:
added comment